x-pack/plugins/alerting/server/routes/rule/apis/find/find_rules_route.ts

```diff
@@ -37,15 +38,12 @@ const buildFindRulesRoute = ({
   router,
   excludeFromPublicApi = false,
   usageCounter,
+  options: routerOptions,
 }: BuildFindRulesRouteParams) => {
   router.get(
```

Comment:
This call to the router will also be done when calling `findInternalRulesRoute`. This will register the `GET /internal/alerting/rules/_find` which we do not want. Should we add a check like

```
if (path !== INTERNAL_ALERTING_API_FIND_RULES_PATH) { /// rouer.get }
```

?

Reply:
Restructured the `if` statement to be an `if...else` to handle this